astropy/modeling/tests/test_pickle.py

```diff
@@ -0,0 +1,210 @@
+
+
+@pytest.mark.parametrize("model", math_functions_all)
+def test_pickle_math_functions(model):
+    m = getattr(math_functions, model)()
+    m1 = loads(dumps(m))
+    if m.n_inputs == 1:
+        assert_allclose(m(x_math), m1(x_math))
+    else:
+        assert_allclose(m(x_math, y_math), m1(x_math, y_math))
+
+
+@pytest.mark.skipif(not HAS_SCIPY, reason="requires scipy")
+def test_pickle_other():
+    # Test models which don't fit in the other test functions.
+
+    x = 0.5
+    y = 1
+
+    model = mappings.Mapping((1, 0))
+    modelp = loads(dumps(model))
+    assert_allclose(model(x, y), modelp(x, y))
+
+    model = mappings.Identity(2)
+    modelp = loads(dumps(model))
+    assert_allclose(model(x, y), modelp(x, y))
+
+    model = mappings.UnitsMapping(((u.m, None),))
+    modelp = loads(dumps(model))
+    assert_allclose(model(x * u.km), modelp(x * u.km))
+
+    model = ArctanhUfunc()
+    modelp = loads(dumps(model))
+    assert_allclose(model(x), modelp(x))
+
+    model = rotations.Rotation2D(23)
+    modelp = loads(dumps(model))
+    assert_allclose(model(x, y), modelp(x, y))
+
+    model = tabular.Tabular1D(lookup_table=[1, 2, 3, 4])
+    modelp = loads(dumps(model))
+    assert_allclose(model(y), modelp(y))
+
+    model = tabular.Tabular2D(lookup_table=[[1, 2, 3, 4], [5, 6, 7, 8]])
+    modelp = loads(dumps(model))
+    assert_allclose(model(x, y), modelp(x, y))
+
+    model = spline.Spline1D()
+    modelp = loads(dumps(model))
+    assert_allclose(model(x), modelp(x))
+
+    model = projections.AffineTransformation2D(
+        matrix=[[1, 1], [1, 1]], translation=[1, 1]
+    )
+    model.matrix.fixed = True
+    modelp = loads(dumps(model))
+    assert_allclose(model(x, y), modelp(x, y))
+    assert model.matrix.fixed is True
+
+
+@pytest.mark.parametrize("model", physical_models.__all__)
+def test_pickle_physical_models(model):
+    m = getattr(physical_models, model)()
+    m1 = loads(dumps(m))
+    if m.n_inputs == 1:
+        assert_allclose(m(x), m1(x))
+    else:
+        assert_allclose(m(x, y), m1(x, y))
+
+
+def test_pickle_polynomial():
+    # models initialized with 1 degree
+    models2d = ["Chebyshev2D", "Hermite2D", "Legendre2D", "InverseSIP"]
+    # models initialized with 2 degree
+    models1d = ["Chebyshev1D", "Hermite1D", "Legendre1D", "Polynomial1D"]
+
+    sip = ["SIP", "InverseSIP"]
+
+    for model in models1d:
+        m = getattr(polynomial, model)
+        m = m(2)
+        m1 = loads(dumps(m))
+        assert_allclose(m(x), m1(x))
+
+    for model in models2d:
+        m = getattr(polynomial, model)
+        m = m(2, 3)
+        m1 = loads(dumps(m))
+        assert_allclose(m(x, y), m1(x, y))
+
+    # Polynomial2D is initialized with 1 degree but
+    # requires 2 inputs
+    m = polynomial.Polynomial2D
+    m = m(2)
+    m1 = loads(dumps(m))
+    assert_allclose(m(x, y), m1(x, y))
+
+    m = polynomial.SIP
+    m = m((21, 23), 2, 3)
+    m1 = loads(dumps(m))
+    assert_allclose(m(x, y), m1(x, y))
```

Comment:
This can be split into 3 tests:

1. A parameterized test over the 2D polynomials
2. A parameterized test over the 1D polynomials
3. The last two special cases.